File: 6 kyu/Binary Tree Compare/BinaryTreeCompare.c
#include <stdbool.h>
#include <stdio.h>

typedef struct TreeNode {
    int val;
    struct TreeNode *left, *right;
} TreeNode;

bool equal_trees (const TreeNode* a, const TreeNode* b)
{
  if (a == NULL || b == NULL) return a == b;
  return (a->val == b->val) && (equal_trees(a->left, b->left) && equal_trees(a->right, b->right));
}

File: 6 kyu/Binary Tree Compare/BinaryTreeCompare.js
function compare(a, b){
  if ((a == null && b != null) || (a != null && b == null)) return false;
  if (a == null && b == null) return true;
  return a.val == b.val && compare(a.left, b.left) && compare(a.right, b.right)
}
